avail-server/src/main/kotlin/com/avail/server/io/AvailServerChannel.kt

```diff
@@ -49,12 +49,28 @@ import java.util.concurrent.atomic.AtomicLong
  *
  * @author Todd L Smith &lt;todd@availlang.org&gt;
  * @author Richard Arriaga &lt;rich@availlang.org&gt;
+ *
+ * @constructor
+ * Construct an [AvailServerChannel].
+ *
+ * @param closeAction
+ *   The custom action that is to be called when the this channel is closed in
+ *   order to support implementation-specific requirements after the closing of
+ *   this channel.
  */
-abstract class AvailServerChannel : AutoCloseable
+abstract class AvailServerChannel constructor(
+	closeAction: (DisconnectReason, AvailServerChannel) -> Unit)
```

Comment:
I'm okay with this one not having a default, so long as every call site passes a real function.